mlir/test/lib/Dialect/Affine/TestReifyValueBounds.cpp

```diff
@@ -84,6 +84,13 @@ static LogicalResult testReifyValueBounds(FunctionOpInterface funcOp,
     auto boundType = op.getBoundType();
     Value value = op.getVar();
     std::optional<int64_t> dim = op.getDim();
```

Comment:
`dim` must be provided if and only if `value` is a shaped type. We should also error out for IR such as:
```
%r = "test.reify_bound"(%int_val) {dim = 1 : i64} : (index) -> index
```